controllers/nodemaintenance_controller.go

```diff
@@ -414,7 +414,7 @@ func (r *NodeMaintenanceReconciler) initMaintenanceStatus(nm *nodemaintenancev1b
 			return err
 		}
 		nm.Status.TotalPods = len(podlist.Items)
-		err = r.Client.Status().Update(context.TODO(), nm)
+		err = r.Client.Status().Update(context.Background(), nm)
```

Comment:
Good to fix the context right now, but then everywhere please. Looks like you need to pass the context around a bit more 🙂